resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/config.scala

```diff
@@ -269,12 +269,41 @@ package object config extends Logging {
       .createWithDefaultString("1s")
 
   // Spark resource staging server.
+  private[spark] val RESOURCE_STAGING_SERVER_API_SERVER_URL =
+    ConfigBuilder("spark.kubernetes.resourceStagingServer.apiServer.url")
+      .doc("URL for the Kubernetes API server. The resource staging server monitors the API" +
+        " server to check when pods no longer are using mounted resources. Note that this isn't" +
+        " to be used in Spark applications, as the API server URL should be set via spark.master.")
+      .stringConf
+      .createWithDefault(KUBERNETES_MASTER_INTERNAL_URL)
+
+  private[spark] val RESOURCE_STAGING_SERVER_API_SERVER_CA_CERT_FILE =
+    ConfigBuilder("spark.kubernetes.resourceStagingServer.apiServer.caCertFile")
+      .doc("CA certificate for the resource staging server to use when contacting the Kubernetes" +
+        " API server over TLS.")
+      .stringConf
+      .createOptional
+
   private[spark] val RESOURCE_STAGING_SERVER_PORT =
     ConfigBuilder("spark.kubernetes.resourceStagingServer.port")
       .doc("Port for the Kubernetes resource staging server to listen on.")
       .intConf
       .createWithDefault(10000)
 
+  private[spark] val RESOURCE_STAGING_SERVER_RESOURCE_TTL =
+    ConfigBuilder("spark.kubernetes.resourceStagingServer.resourceTtl")
+      .doc("Time for which resources should remain on the resource staging server before they" +
+        " cleaned up. Note that this is the time period after the resource staging server first" +
+        " detects that no pods are present that are using their mounted resources.")
+      .timeConf(TimeUnit.MILLISECONDS)
+      .createWithDefaultString("30m")
```

Comment:
So this should reflect the amount of time we think it would take for pods to restart upon failure and for pods to want to re-retrieve those resources after crashing, as opposed to the amount of time the resources should remain available in general.